plugins/algebra/array/include/traccc/plugins/algebra/array_definitions.hpp

```diff
@@ -23,6 +23,8 @@
 #include <map>
 #include <tuple>
 
+#define ALGEBRA_PLUGIN detray::array
```

Comment:
Curious. Where is `ALGEBRA_PLUGIN` used? You added these macros, but I don't see any new code that would use them.

I anyway wonder, even if we need these macros, whether it would not be better to define them in the build system. 🤔 But first let's make sure that we actually need them.

Reply:
In detray I removed it, but traccc still uses a ```default_algebra``` in many cases, which is defined here
https://github.com/acts-project/traccc/blob/862076927a59bdfc969cea0ebd10cfea00ee7391/core/include/traccc/definitions/primitives.hpp#L33

Reply:
Hmm... Let's keep the macro in this PR then. But I'll ponder a bit about what best to do in a follow-up. As even with global type definitions, we should still not use macros like this here. 🤔